wave_kinematics/src/rotation.cpp

```diff
@@ -0,0 +1,99 @@
+#include "wave/kinematics/rotation.hpp"
+#include "wave/utils/utils.hpp"
+
+double wave::Rotation::comparision_threshold = 1e-8;
+
+wave::Rotation::Rotation() {}
+
+void wave::Rotation::setToIdentity() {
+  this->rotation_object.setIdentity();
+}
+
+void wave::Rotation::compose(const Rotation &R) {
+  this->rotation_object = this->rotation_object * R.rotation_object;
+}
+
+wave::Mat3 wave::Rotation::invert() {
+  this->rotation_object.invert();
+}
+
+wave::Vec3 wave::Rotation::rotate(const Vec3 &input_vector) {
+  return this->rotation_object.rotate(input_vector);
+}
+
+wave::Vec3 wave::Rotation::inverseRotate(const Vec3 &input_vector) {
+  return this->rotation_object.inverseRotate(input_vector);
+}
+
+wave::Vec3 wave::Rotation::rotateAndJacobian(const Vec3 &input_vector,
```

Comment:
Not sure I understand. Is "used member functions " a new change to the code in this PR, or are you explaining your original choice?

To me, calls of the form
```cpp
auto rotated_vector = someRotateFunction(input_vector);
auto jacobian = someJacobian(input_vector);
auto other_jacobian = anotherJacobian(input_vector);
```
are more clear than
```cpp
Mat3 jacobian; // uninitialized
Mat3 other_jacobian; // uninitialized
 // Note: can't tell which parameter is which without looking it up
Vec3 rotated_vector = rotateAndJacobian(input_vector, jacobian, other_jacobian);
```